Fix overpass speed being detected

SippieCup opened the pull request.

pd0wm committed:
Revert dPoly computation to 0.6.3 behavior (#809)

* Revert dPoly computation to 0.6.3 behavior

* update test reference

rbiasini committed:
Mistakenly removed Honda Fit support from README.md

pd0wm committed:
Add spinner source files (#812)

illumiN8i committed:
2020 Toyota Prius Prime Limited (#813)

* 2020 Prius Prime Limited

Fingerprint for 2020 Prius Prime Limited from garglo. 86a0adcb192c9424

* Update README.md

2020 supported

rafcabezas reviewed (dismissed):
Looks good to me, should we bother with an abs() here to capture the opposite scenario, but still delta of >20mph?

kishkaru committed:
Update thermald to set ignition_seen back to False if Panda disconnected (#820)

illumiN8i committed:
Add 2020 Corolla XSE AFS option (#821)

Frank McTank on Discord has XSE with Connectivity Package with Adaptive Front Lighting System option

SippieCup commented:
We dont want to abs() because sometimes the speed limit is only 5 and apperently thats the default for tesla, so it'll ignore most speed limits when it gets to that point.

That said, I think a better solution like @bitshop said is the right approach. Unfortunately, I don't think the map data gives a direction for the lane you are in. So we can't go off that. we need some memory of previous speed limits at different ticks, and only act after a certain number?

csouers committed:
Add 2018 Civic Hatchback 1.0l CVT (European) support (#823)

* Add 2018 Civic Hatchback 1.0l CVT support

test_fingerprints returns fine

neon-dev commented:
@SippieCup if I'm not mistaken the var naming is misleading. It looks like `pedal_set_speed_ms` is the current pid speed and `speed_limit_ms` is max pedal speed.
I didn't test it but in that case this change would not work as expected. Also I don't know how 46.3 m/s matches with the comment stating 20 mph.
And lastly this commit only affects fleet speed and not speed limit detection. Was this intentionally?
Keep in mind that there are countries like Germany with no speed limit (currently capped in code at 170 kph), which may also cause corner cases.

Commit:
Merge cereal subtree

Commit:
Squashed 'cereal/' changes from ea14abe4b..3d90c7877

3d90c7877 Fix wrong event enum
f4c92cf4e add none/invalid usb power mode
12bc18965 Add fields and states required for robust Volkswagen safety compliance (#8)
16b5850f1 let's spell wolkswagen
6243622e6 Safety cereal (#9)
5cf3c2c76 change order of UsbPowerMode to preserve panda lib behavior
18a291779 usbPowerMode is more useful to log and it comes from panda
152678429 add ldw to visual hud alerts (#5)
24f6fa78c merge https://github.com/commaai/cereal/pull/3 after fixing conflicts
4d5862c54 added event about internet connection
ca070f9fd Added usbPowerOn to health
e2eaa762b add HW type for UNO
e528a2041 Read RPM from fan connected to Panda
8472175d6 Added invalidGiraffeToyota event
71be97053 placeholders for mazda, nissan and vw safety models

git-subtree-dir: cereal
git-subtree-split: 3d90c7877f9ef9269735b074c0cb23f5eb6d2b47

Commit:
Squashed 'panda/' changes from 9881e6118..30c7ca8a5

30c7ca8a5 bump version to 1.5.3
9403dbebe Need to fix wifi test before re-enabling.
0812362b5 GPS UART fix until boardd is refactored (#294)
ffbdb87a8 python2 -> 3 fixes to pedal flasher (#292)
78b75ef59 Added build type to release version strings
736c2cbf7 Fixed sending of bytes over PandaSerial
0894b28f1 Fixed USB power mode on black (#291)
4b3358c92 patch to be able to switch from EON to PC with a Panda that has EON b… (#290)
a95c44a71 Made setting of NOOUTPUT on no heartbeat more efficient (#287)
948683688 UART instability fix with high interrupt load (#283)
9a9e9d47b Fix usb_power_mode missing initialization (#289)
af0960ad3 DFU fix (#288)
70219d7bb match safety enum in cereal (#285)
a338d3932 Fix build for jenkins test
78ef4a6eb Stop charge (#284)
5266a4028 Fix typo (#286)
f4787ec5a Revert "turn on CDP when ignition switches on (#281)"
d37daee97 Revert "NONE and CLIENT should be the same thing in white/grey pandas"
e97b283e7 NONE and CLIENT should be the same thing in white/grey pandas
8c1df559f turn on CDP when ignition switches on (#281)
847a35d42 Fix bullet points
fac027716 Misra update (#280)
5a04df6b1 Added description of regression tests to README
c4aabae59 Fixed some python3 bugs in the test scripts and PandaSerial
9af0cb353 Bump version
c4ac3d63b Disable GPS load switching on black pandas
078ee588c This is the correct table, actually
578b95ee3 Misra table of coverage added
d383a2625 bump panda
b98ca010d fix sdk build in python3 env (#279)
63d3dc7d3 Set python3 env before runnign get_sdk, so we know if it fails
e951d79c0 legacy code we don't control can remain python2
11b715118 Merge pull request #276 from commaai/python3
9893a842a Merge pull request #277 from zorrobyte/patch-1
d3268690c Revert "revert back esptool to python2 and force to build esptools with python2"
875e76012 revert back esptool to python2 and force to build esptools with python2
9c40e6240 needed to install python3
ed2ac87cf Also moved safety tests to python3
6842b2d2c move esptool sdk installation before python3 env is set. Kind of a cheat
b5a2cabcd this hopefully fixes build test
628050955 Fixes safety replay
2c220b623 this fixes language regr test
fdbe789b8 use python 3 in Docker container
ee1ae4f86 Better hash print
0de9ef73c Revert "Final 2to3 on the whole repo"
c92fd3bc9 Final 2to3 on the whole repo
5f2bc4460 better
b2a30fdbd make works!
b74005d10 fix sign.py
fe727706b read file as byte and no tab before sleep
32a344ef6 Update README.md
2dc34096a 2to3 applied
ffa68ef71 undo unnecessary brackets for print
dbc248027 Fix all the prints with 2to3, some need to be undo
5a7aeba0f xrange is gone
982c4c928 one more python3 env
1e2412a29 env python -> env python3

git-subtree-dir: panda
git-subtree-split: 30c7ca8a53a3adb05d23d7cfe64fb716a656ef1a

Commit:
Merge panda subtree

Commit:
Merge opendbc subtree

Commit:
Squashed 'opendbc/' changes from 684e28a7a..f3b573559

f3b573559 move generator to python3
9efff4086 2019+ New Prius Steer Angle (#189)
f3461d143 add units and a couple new signals for toyota (#188)
16033b3ff Remove non ascii characters
de162d8c2 Toyota no dsu: fix steer angle factor, it's 1% of a rad
2fb524554 Lexus CT200H seems to have the safetyParam 1 instead of 0.73
b2fde8b11 toyota time signal (#187)
6cfe0c432 Fixed brake signal unit in Bosch Honda

git-subtree-dir: opendbc
git-subtree-split: f3b573559f1619346f082554a72cebcc43da5275

Commit:
openpilot v0.6.5 release

BogGyver committed:
first pass at 0.6.5 merge

BogGyver committed:
most python fixes

BogGyver committed:
fix readconfig and panda code

BogGyver committed:
final pre car_test

BogGyver committed:
radar and radartools fix

BogGyver committed:
some fixes, not all

BogGyver committed:
radar_interface fixes

BogGyver committed:
first drive try

BogGyver committed:
fix for button status

BogGyver committed:
PCC now engages, not tried on road

BogGyver committed:
ACC fix

BogGyver committed:
fix for UIBT_module.py

SippieCup committed:
MagicNumberz for tuning

Reduce EonToFront

Fix overpass speed being detected
